WebDevStudios/wds-block-starter
# Feature/#9 postcss config

## Closes #9 

### Description
- Setup CSS Nano to apply on production builds and minify styles.
- Add PostCSS Safe Parser to prevent incorrect CSS from being transcoded into production builds.
- Setup Media Query Packer to extract and organize media queries throughout styles.
- Add Node Sass in place of Precss. Note: While precss is favorable for build speed and features it does not support traditional Sass syntax at this time for certain features.
- Add sass-loader to webpack config and install node-sass package along with sass-loader.
- Uninstall precss releated packages.
- Update postcss.config and remove out precss related requirements.

### How to Test
Reviewers can test this feature by checking out this branch locally and using `npm run dev` and `npm run build` to inspect files. Optionally you can also use the test branch provided below.

### Test Branch
I setup test examples in `style.scss` on [example/#9-postcss-test](https://github.com/WebDevStudios/wds-block-starter/tree/example/%239-postcss-test).

### Features to Test
1. Sass functionality works correctly
2. Using `npm run dev` build files work properly
2. Using `npm run build` production files build as expected
3. Verify that media queries group correctly
4. Verify that incorrectly written styles do not make it's way into built files
